oracle.py

```diff
@@ -214,7 +234,10 @@ def generate_resources(path=""):
     for crd in crd_list:
         resources[crd] = get_crd(crd)
 
-    trim_resource(resources)
+
+    if learn_twice and data_dir != "":
+        base_resources = json.loads(open(os.path.join(data_dir, "resources.json")).read())
```

Comment:
Pls correct me if I am wrong, I guess we do not need to check both `learn_twice` and `data_dir`?
As long as `learn_twice` is true, we should have the valid data_dir here? Similarly I guess we do not need to give data_dir an empty string default value?

Reply:
I feel that maybe we should not store the resources learnt from `learn-once` into `data_dir`, as every time after a new run there would be many changes to the "resources.json" (like the timing). Maybe you should just store the resource.json in `operator_name/test_name/learn/learn-once` in the first run, and in the second run you compare it and the new one in `operator_name/test_name/learn/learn-twice`

Reply:
So here the data_dir seems redundant if we directly fetch learn once log from the `log_dir`. But in that case, there is an assumption for the dir structure, like if we are current in `learn_twice` log dir, the dir for `learn-once` should be `learn_twice/../learn_once`, which doesn't seem very elegant for me, but anyway, I have changed to that way.